gpu/octree/src/cuda/octree_host.cu

```diff
@@ -176,6 +176,23 @@ void pcl::device::OctreeImpl::radiusSearchHost(const PointType& query, float rad
             end = beg + std::min<int>((int)out.size() + end - beg, max_nn) - (int)out.size();
 
             out.insert(out.end(), host_octree.indices.begin() + beg, host_octree.indices.begin() + end);
+
+            if (find_dist)
```

Comment:
if `find_dist` is not true, you should fill `sqr_distances` with nan for the points inside this node. It's important to ensure `sqr_distances` has the same size of `out`